src/alire/os_linux/alire-platform.adb

```diff
@@ -34,30 +35,90 @@ package body Alire.Platform is
             use Utils;
             Release : constant Utils.String_Vector :=
                         Subprocess.Checked_Spawn_And_Capture
-                          ("cat", Empty_Vector & "/etc/os-release");
+                ("cat", Empty_Vector & "/etc/os-release");
+
+            function Get_Os_Release_Value_For_Key (Key : String;
+                                                   Multiple_Values : Boolean)
+                                      return Alire.Platforms.Distributions is
+
+               use GNAT.Regpat;
+
+               --  Regexp accepting lines not starting with '#' like:
+               --  key=value
+               --  key='value'
+               --  key='value1 value2'
+               --  key="value"
+               --  key="value1 value2"
+               Regexp : constant Pattern_Matcher :=
+                 Compile ("^" & Key & "=[""']?([^""']+)[""']?");
+               Matches : Match_Array (1 .. 1);
+
+            begin
+               for Line of Release loop
+                  declare
+                     Normalized : constant String :=
+                       To_Lower_Case (Line);
+
+                     Values : String_Vector;
+                  begin
+                     Match (Regexp, Normalized, Matches);
+                     if Matches (1) /= No_Match then
+                        if Multiple_Values then
+                           --  Generate Values from space separated items
+                           Values :=
+                             Split (
+                               Normalized
+                                 (Matches (1).First .. Matches (1).Last), ' ');
+
+                        else
+                           --  Generate Values from a single value
+                           Values :=
+                             To_Vector (
+                               Normalized
+                                 (Matches (1).First .. Matches (1).Last));
```

Comment:
Given that, according to https://www.freedesktop.org/software/systemd/man/os-release.html, ID cannot contain spaces, I'd say this `Multiple_Values` isn't necessary, and splitting on spaces can be always attempted, which will simplify this snippet a bit.

Reply:
Ok, to be honest I was a bit afraid to always try to split ID values.
But you are right : I will remove the distinction and simplify the processing.